### ISSUE TITLE
Use .env file for Docker Compose Port

### ISSUE THREAD

**Issue by Pro** [loft-sh/devpod#531]
**What happened?**

Running devpod with the Docker Provider failed with:

> \[15:57:30\] info 1 error(s) decoding:
> \[15:57:30\] info \* error decoding 'Ports': No port specified: :<empty>

When using a docker-compose.yaml file with:

```
version: '3'
services:
  my-service:
    container_name: some-container
    restart: on-failure:3
    pull_policy: if_not_present
    ports:
      - ${MY_PORT}:${MY_PORT}
```

and a `.env` file with:

```
MY_PORT=80
```

**What did you expect to happen instead?**

DevPod should correctly run docker compose and consider the .env file in the same directory to replace variables used inside the docker compose file.

**How can we reproduce the bug?** (as minimally and precisely as possible)

My `devcontainer.json`:

```
{
    "name": "my-dev",
    "context": ".",
    "dockerComposeFile": [
        "docker-compose.yaml"
    ]
}
```

**Local Environment:**

* DevPod Version: 0.2.1
* Operating System: linux
* ARCH of the OS: AMD64

**DevPod Provider:**

* Local/remote provider: docker

**Anything else we need to know?**

I already tested putting the .env file into the repo root and various other places, but no success.
With docker compose it depends on the working directory used to run compose files. I guess DevPod is not using the .devcontainer directory, nor the repo root to run docker compose?

Note that the exact same setup works quite fine when using VSCode directly.

**Comment by Pro** [comment]
Additional note:

If I first use the following command to create a docker compose config with all variables replaced:

```
docker compose -f docker-compose.yaml config > docker-compose.config.yaml
```

and then put the `docker-compose.config.yaml` into the devcontainer.json config, devpod is able to start the container.

**Comment by Pro** [comment]
Thanks for the quick answer and fix! I can confirm that this is fixed in 0.3.0
